corral/cmd/cmd_fetch.pony

```diff
@@ -4,59 +4,83 @@ use "../bundle"
-    match BundleFile.load_bundle(ctx.env, ctx.directory, ctx.log)
-    | let bundle: Bundle =>
-      fetch_bundle_deps(bundle, bundle)
+    match BundleFile.load_bundle(ctx.bundle_dir, ctx.log)
+    | let bundle: Bundle val =>
+      _Fetcher(ctx, bundle).fetch_bundle_deps(bundle)
     | let err: Error =>
       ctx.env.out.print(err.message)
       ctx.env.exitcode(1)
     end
 
-  fun fetch_bundle_deps(base_bundle: Bundle box, bundle: Bundle) =>
-    ctx.log.info("Fetching bundle: " + bundle.name())
+actor _Fetcher
+  let ctx: Context
+  let base_bundle: Bundle val
+
+  new create(ctx': Context, base_bundle': Bundle val) =>
+    ctx = ctx'
+    base_bundle = base_bundle'
+
+  be fetch_bundle_deps(bundle: Bundle val) =>
+    ctx.log.info("Fetching direct deps of bundle: " + bundle.name())
     for dep in bundle.deps.values() do
-      try
+      //try
         ctx.log.info("Fetching dep: " + dep.name() + " @ " + dep.version())
           //dep.data.locator)
-        fetch_dep(base_bundle, dep)?
-      else
-        ctx.log.err("Error fetching dep: " + dep.name() + " @ " + dep.version())
-          //dep.data.locator)
-      end
+        fetch_dep(dep)
+      //else
+      //  ctx.log.err("Error fetching dep: " + dep.name() + " @ " + dep.version())
+      //    //dep.data.locator)
+      //end
     end
-    for dep in bundle.deps.values() do
-      // TODO: prevent infinite recursion by keeping & checking a deps Set
-      try
-        let bundle_dir = base_bundle.dep_bundle_root(dep)?
-        ctx.log.fine("Fetching dep's bundle in: " + bundle_dir.path)
-        let dep_bundle = Bundle.load(ctx.env, bundle_dir, ctx.log)?
-        ctx.log.fine("Fetched dep's bundle is: " + dep_bundle.name())
-        fetch_bundle_deps(base_bundle, dep_bundle)
-      else
-        ctx.log.err("Error loading/fetching dep's bundle: " + dep.flat_name())
-      end
+    //fetch_transitive_deps(bundle)
+
+  be fetch_transitive_dep(dep: Dep val) =>
+    ctx.log.info("Fetching transitive dep: " + dep.name())
+    // TODO: prevent infinite recursion by keeping & checking a deps Set
+    try
+      let bundle_dir = base_bundle.dep_bundle_root(dep)?
+      ctx.log.fine("Fetching dep's bundle into: " + bundle_dir.path)
+      let dep_bundle: Bundle val = Bundle.load(bundle_dir, ctx.log)?
+      ctx.log.fine("Fetched dep's bundle is: " + dep_bundle.name())
+      fetch_bundle_deps(dep_bundle)
+    else
+      ctx.log.err("Error loading/fetching dep bundle: " + dep.flat_name())
+      ctx.env.exitcode(1)
     end
 
-  // TODO:
-  // We want the fetch workspace to always be directly under this project's
-  // corral, but we still locate the fetched bundles from down in corral.
+  fun fetch_dep(dep: Dep val) =>
+    try
+      let local = ctx.repo_cache.join(dep.flat_repo())?
+      let workspace = base_bundle.dep_workspace_root(dep)?
+      let repo = Repo(dep.repo(), local, workspace)
+      let vcs = VCSForType(ctx.env, dep.vcs())?
+
+      // TODO: deal with versions
```

Comment:
is there an issue open for this?

Reply:
Yes, this is https://github.com/ponylang/corral/issues/59

Reply:
Can you add a reference(s) to the issues that the TODO's are part of so that others could, in theory, pick up the work and have the TODO's be pointers for said issues?